ansible_collections/arista/avd/docs/contribution/authoring_eos_cli_config_gen.md

```diff
@@ -0,0 +1,111 @@
+9. Along with EOS config template, update the documentation template as well (if required).
+10. Implement only commands visible in `show running-config all` of the EOS device. We should not hide config if given by the user.
+11. Validate the template using j2lint tool, run `pre-commit run j2lint --all`.
+
+### Build Schemas and Documentation
+
+Run `pre-commit run schemas --all` to re-generate the eos_cli_config_gen schema with any modifications. This command should be executed every time the schema is changed, even if only a description is updated.
+It also updates the documentation with new options.
+
+### Add Molecule Tests
+
+Add some molecule tests in the `ansible_collections/arista/avd/molecule/eos_cli_config_gen` scenario exercising the new knob configuration.
+
+When marking any key as "deprecated", move the related tests to the `eos_cli_config_gen_deprecated_vars` molecule scenario and add any missing tests if necessary.
+
+### Run Molecule Tests
+
+Run `molecule converge -s eos_cli_config_gen` from the path `ansible_collections/arista/avd/` to execute the molecule tests locally and generate the new expected configuration and documentation for newly added test-cases.
+
+Check the PyAVD test coverage report by running `tox -e coverage,report` and work on improving the coverage where possible.
+
+### Update Documentation
+
+If the proposed feature requires any changes to the documentation, make sure to update it accordingly.
+If there are any breaking changes introduced, document them in the porting guide.
+
+### Run Pre-commit Checks
+
+Run all pre-commit checks `pre-commit run --all` to ensure that all files added or modified are correctly following the coding standards and formatting rules.
+Running these checks also ensures that the changes pass CI checks.
+
+### Self Review The Changes
+
+Before pushing the changes to GitHub, carefully review all the modifications.
+Confirm that all new features work as intended and that existing features are unaffected. Once satisfied, proceed to push the changes to the repository.
+
+## Checklist to review an eos_cli_config_gen PR
+
+1. Check that all the CI checks are passing.
+2. If the PR addresses an issue raised in the repository, confirm that the issue is fully resolved by the PR.
+3. Refer to the Arista documentation for a deeper understanding of the proposed feature.
+4. Verify that the schema adheres to EOS CLI and all relevant guidelines mentioned above.
+5. Ensure that the min/max and valid-values are specified in the schema if they are defined in the EOS CLI.
+6. Ensure that Jinja2 templates follow all the guidelines mentioned above.
+7. Check that the template generates valid configuration and documentation, maintaining the same configuration order and indentation as EOS CLI.
+8. Check out the PR in a local IDE and run all tests to ensure functionality.
+9. Test the generated configuration through EAPI or CVP in the ATD lab.
```

Comment:
not everybody may have an ATD lab - you can state `in a relevant lab (if availbale otherwise indicate you could not verify in a lab in your review comment.)`
```suggestion
9. Test the generated configuration through EAPI or CVP in the ATD lab.
```

Reply:
Updated